### ISSUE TITLE
Hydrostatic condition for open domain

### ISSUE THREAD

**Issue by mameehan5** [AMReX-Combustion/PeleLMeX#91]
Hi all, 

I am looking to run a buoyancy-driven flow that is open to atmospheric conditions. However, when running a simulation with boundary conditions specified as "Outflow" normal to the direction of gravity, the hydrostatic condition induced by the gravitational field disappears.  

To show this most clearly, starting from the HotBubble tutorial case, the following conditions 
```
geometry.is_periodic = 1 0               # For each dir, 0: non-perio, 1: periodic
peleLM.lo_bc = Interior NoSlipWallAdiab
peleLM.hi_bc = Interior Outflow
```
produced an `avg_pressure` field (using `yt`)
<img src="https://user-images.githubusercontent.com/36512136/170776319-e5122ca5-8945-42e7-9fe3-21096ac6b4e0.png" width="200" height="200">
as expected for the first snapshot with a hydrostatic condition.

Using
```
geometry.is_periodic = 0 0               # For each dir, 0: non-perio, 1: periodic
peleLM.lo_bc = NoSlipWallAdiab NoSlipWallAdiab
peleLM.hi_bc = NoSlipWallAdiab Outflow
```
produced
<img src="https://user-images.githubusercontent.com/36512136/170776363-e4a7d881-58ea-4900-a40d-ba346a9af4a8.png" width="200" height="200">
Again, this is expected.

And finally, using
```
geometry.is_periodic = 0 0               # For each dir, 0: non-perio, 1: periodic
peleLM.lo_bc = Outflow NoSlipWallAdiab
peleLM.hi_bc = Outflow Outflow
```
produced
<img src="https://user-images.githubusercontent.com/36512136/170776400-c2ca7e31-4cb3-401d-8c51-de4437abd55c.png" width="200" height="200">

This last image I would have expected to be quite similar to the other two. Is there a physical justification for this behavior or a bug? If the former, is there a way to introduce the hydrostatic condition?

Thank you very much!!

**Comment by mameehan5** [comment]
I realized now that I can just force the hydrostatic condition with `gradP0` for fix my personal simulations. However, I still would like confirmation this is the expected behavior before closing, so please let me know!

**Comment by esclapez** [comment]
That rings a bell with an old issue of PeleLM that @nickwimer found a while back.
I'll see if the solution implemented in:
https://github.com/AMReX-Combustion/IAMR/commit/9f5f52de343d7b751fac658cfdbd7e815ca98198
can fix the issue.

**Comment by mameehan5** [comment]
Sounds great! Those snippets do seem like they would do the trick.

**Comment by nickwimer** [comment]
Just checking in to see if this issue has been resolved?

**Comment by esclapez** [comment]
Not yet, the entire infrastructure to handle the phi BC isn't setup in LMeX. I'm thinking it could actually live into AMReX-Hydro projectors, but I need to discuss that with the LBNL team.

**Comment by Alex-jian522** [comment]
Is this problem solved now?

**Comment by mameehan5** [comment]
For myself, I just set `gradP0` for the hydrostatic condition I need. This seems to work well, and in fact, may even be better than what I posted as the problem because this is no ambiguity as to what the ambient density would be. I will let the developers decide whether they want to pursue this further or close the issue.

**Comment by nickwimer** [comment]
I am in the middle of a rewrite to sections of the code to fix the underlying issue at play here. In the meantime, @mameehan5  would you be willing to share your solution involving the `gradP0`? It might be a good "band-aid" solution that we can patch for others facing the same type of issue in the short term.

**Comment by mameehan5** [comment]
Yep! In the input file, if using gravity, say `peleLM.gravity = 0 0 -9.81`, I define `peleLM.gradP0` to be g*rhoinf where rhoinf is the ambient density. So then buoyancy is only generated where there is a density difference (i.e., density different than the ambient).

**Comment by drummerdoc** [comment]
Did this get resolved @nickwimer ?

**Comment by d-montgomery** [comment]
Adding another example case for this issue that was solved by setting $\vec{\nabla}p_0 = \rho_0\vec{g}$, where $\rho_0$ is the initial density at the inlet.  The set up is as follows: we use embedded boundaries to define a 3D cylindrical pipe with `TurbInflow` to simulate turbulent flow in a pipe.  The pipe has `noSlipIsotherm` walls that decrease in temperature from the inlet to the outlet, with inflow and outflow boundary conditions.  The figure below demonstrates the effect of gravity with and without setting $\vec{\nabla}p_0 = \rho_0\vec{g}$. 

![Image](https://github.com/user-attachments/assets/b551f553-1acb-401a-816f-28ec942faf49)
